simulations/src/vehicle/vehicle.py

```diff
@@ -0,0 +1,80 @@
+class Vehicle:
+
+    t_c = 10  # number of steps for period of communication (10ms*t_c = 100ms)
+    max_speed = 100 # (km/h)
+    max_acceleration = 0.05 # speed (km/h) / step
+    max_deceleration = 0.2 # speed (km/h) / step
+    array_distance_errors_len = 100
+
+    def __init__(self, order) -> None:
+        self.speed = 0  # initially it the vehicle stands still
+        self.speed_old = 0  # updates each period t_c
+        self.position = 0  # current position of the vehicle in meters
+        self.distance = 0  # distance to the vehicle in front (-1 if no one is in front)
+        self.order = order  # number of vehicles in front (first has order = 0)
+        self.array_distance_errors = [0]*self.array_distance_errors_len
+        self.array_distance_errors_pointer = 0
+        self.error_derivative = 0
+        self.prev_distance_error = 0
+        self.min_distance = 9  # how close the vehicles should be to each other, depends on speed. (m)
+
+    # This should be called each step
+    def update_speed(self, step):
```

Comment:
Denna funktionen verkar öka hastigheten varje step. Är det verkligen önskat? Stämmer funktionsnamnet då? Borde den inte heta update_speed_half_acceleration?

Reply:
update_speed should be implemented by each child class. So this should be removed